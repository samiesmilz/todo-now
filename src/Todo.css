.Todo {
  border: 1px solid rgba(0, 0, 0, 0.377);
  text-align: left;
  background-color: aquamarine;
}

.Todo-item {
  padding: 5px;
}
.Todo-controls {
  display: inline;
  margin: 0px 15px;
}

.Todo-input {
  padding: 5px;
  margin-right: 5px;
}
