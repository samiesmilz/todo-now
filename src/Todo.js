import React, { useState } from "react";
import "./Todo.css";

const Todo = ({ item, remove, update, id }) => {
  const [isEditing, setIsEditing] = useState(false);
  const [newData, setNewData] = useState(item);
  const handleRemove = () => {
    remove(id);
  };
  const handleUpdate = (evt) => {
    evt.preventDefault();
    if (newData.trim() !== "") {
      update(id, newData);
      setIsEditing(false);
    }
  };

  const handleChange = (e) => {
    setNewData(e.target.value);
  };

  const handleEditRequest = () => {
    setIsEditing(true);
  };

  let todoItem = (
    <div className="Todo">
      <li className="Todo-item">
        {item}
        <div className="Todo-controls">
          <button onClick={handleEditRequest}>edit</button>
          <button onClick={handleRemove}>remove</button>
        </div>
      </li>
    </div>
  );

  if (isEditing) {
    todoItem = (
      <div>
        <form onSubmit={handleUpdate}>
          <input
            className="Todo-input"
            type="text"
            value={newData}
            onChange={handleChange}
          />
          <button>Update!</button>
        </form>
      </div>
    );
  }

  return todoItem;
};

export default Todo;
