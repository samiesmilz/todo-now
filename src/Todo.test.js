import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import Todo from "./Todo";

it("renders without crashing", function () {
  render(<Todo />);
});

it("matches snapshot", function () {
  const { asFragment } = render(<Todo />);
  expect(asFragment()).toMatchSnapshot();
});

it("executes the delete function on button click", function () {
  const removeMock = jest.fn();
  render(<Todo remove={removeMock} />);
  const deleteButton = screen.getByText("remove");
  fireEvent.click(deleteButton);
  expect(removeMock).toHaveBeenCalled();
});
