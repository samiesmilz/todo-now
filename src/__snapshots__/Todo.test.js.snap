// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Todo"
  >
    <li
      class="Todo-item"
    >
      <div
        class="Todo-controls"
      >
        <button>
          edit
        </button>
        <button>
          remove
        </button>
      </div>
    </li>
  </div>
</DocumentFragment>
`;
